WireDomain/Tests/WireDomainTests/LocalStores/UpdateEventsLocalStoreTests.swift

```diff
@@ -0,0 +1,303 @@
+        // Given there are stored envelopes.
+
+        try await insertStoredEventEnvelopes([
+            Scaffolding.envelope1,
+            Scaffolding.envelope2,
+            Scaffolding.envelope3
+        ])
+
+        // When it deletes 2 envelopes.
+
+        try await sut.deleteNextPendingEvents(limit: 2)
+
+        // Then the first 2 envelopes were deleted.
+
+        try await context.perform { [context] in
+            let request = StoredUpdateEventEnvelope.sortedFetchRequest(asending: true)
+            let result = try context.fetch(request)
+
+            XCTAssertEqual(result.count, 1)
+
+            let envelope = try XCTUnwrap(result.first)
+            XCTAssertEqual(envelope.sortIndex, 2)
+
+            let decoder = JSONDecoder()
+            let decodedEnvelope = try decoder.decode(UpdateEventEnvelope.self, from: envelope.data)
+            XCTAssertEqual(decodedEnvelope, Scaffolding.envelope3)
+        }
+    }
+
+    func testStoreLastEventID_It_Stores_Last_Event_Envelope_ID() throws {
+        // Given
+
+        let id = UUID()
+
+        // When
+
+        sut.storeLastEventID(id: id)
+
+        // Then
+
+        let lastEventId = try XCTUnwrap(mockUserDefaults.string(forKey: Scaffolding.lastEventIDUserDefaultsKey))
+        XCTAssertEqual(UUID(uuidString: lastEventId), id)
+    }
+
+    private func insertStoredEventEnvelopes(_ envelopes: [UpdateEventEnvelope]) async throws {
+        try await context.perform { [context] in
+            let encoder = JSONEncoder()
+
+            for (index, envelope) in envelopes.enumerated() {
+                let storedEventEnvelope = StoredUpdateEventEnvelope(context: context)
+                storedEventEnvelope.data = try encoder.encode(envelope)
+                storedEventEnvelope.sortIndex = Int64(index)
+            }
+
+            try context.save()
+        }
+    }
+
+    private enum Scaffolding {
+
+        static let localDomain = "local.com"
+        static let selfUserID = UserID(uuid: .mockID1, domain: localDomain)
+        static let selfClientID = "abcd1234"
+        static let conversationID = ConversationID(uuid: .mockID2, domain: localDomain)
+        static let lastEventID = UUID.mockID3
+        static let otherDomain = "other.com"
+        static let aliceID = UserID(uuid: .mockID4, domain: otherDomain)
+        static let aliceClientID = "efgh5678"
+        static let defaultsTestSuiteName = UUID().uuidString
+
+        static let id1 = UUID.mockID1
+        static let id2 = UUID.mockID2
+        static let id3 = UUID.mockID3
+        static let id4 = UUID.mockID4
+        static let id5 = UUID.mockID5
+
+        static let time30SecondsAgo = Date(timeIntervalSinceNow: -30)
+        static let time20SecondsAgo = Date(timeIntervalSinceNow: -20)
+
+        static let lastEventIDUserDefaultsKey = "\(selfUserID.uuid.uuidString)_lastEventID"
+
+        nonisolated(unsafe) static let envelope1 = UpdateEventEnvelope(
```

Comment:
Why is `nonisolated(unsafe)` needed?

Reply:
removed in https://github.com/wireapp/wire-ios/pull/2141/commits/ff83af996921a21f1496f91da4889d35e7168186, add Sendable conformance to related value types